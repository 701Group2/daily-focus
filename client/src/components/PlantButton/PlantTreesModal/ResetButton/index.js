import React from "react";
import styles from "./style.module.scss";

function ResetButton({ resetPlant, image }) {
    return (
        <button className={styles.button}>
            <img src={image} onClick={resetPlant} className={styles.image} />
        </button>
    );
}

export default ResetButton;
