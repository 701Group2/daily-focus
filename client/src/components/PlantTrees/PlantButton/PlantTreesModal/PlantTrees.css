.App {
    text-align: center;
}
.CloseButton {
    height: 50px;
    width: 50px;
}

.App-header {
    background-color: #30a0f5;
    height: 20px;
    padding: 20px;
    color: white;
}

.Plant-container {
    width: 150px;
    height: 150px;
    margin: 30px auto;
    border: solid black 1px;
}

.Plant-image {
    height: 100%;
    max-width: 200px;
}

.ControlButton-button {
    background-color: transparent;
    height: 50px;
    width: 50px;
    border: solid #30a0f5 5px;
    border-radius: 30px;
    margin: 0 10px;
}

.ControlButton-image {
    width: 100%;
}

.UserFeedback {
    margin: 10px;
}

progress {
    height: 50px;
    width: 300px;
}
.plantButton {
    width: 50px;
    height: 50px;
    background: #30a0f5;
    border-radius: 200px;
    font-weight: 100;
}
