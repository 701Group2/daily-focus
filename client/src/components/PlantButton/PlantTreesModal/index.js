import React, { useState } from "react";
import Plant from "./Plant";
import ControlButton from "./ControlButton";
import ResetButton from "./ResetButton";
import UserFeedback from "./UserFeedback";
import stateUtils from "../stateUtils";

import styles from "./style.module.scss";

import createPersistedState from "use-persisted-state";
const usePlantProgressState = createPersistedState("plantProgress");

function PlantTreesModal({ waterCoins, onSpendPoint }) {
    const [plantProgress, setPlantProgress] = usePlantProgressState(0);

    const POINTS = {
        SEEDLING: 2, //small trees
        SMALL: 5, //medium trees
        FULL_GROWN: 10, // big trees
    };

    //This var shows different trees pictures
    const getPlantSize = () => {
        if (plantProgress >= POINTS.FULL_GROWN) return 3;
        if (plantProgress >= POINTS.SMALL) return 2;
        if (plantProgress >= POINTS.SEEDLING) return 1;
        return 0;
    };

    const [config, setConfig] = useState({
        plantSize: getPlantSize(), //This var shows different trees pictures
        feedback: stateUtils.feedback.start,
    });

    const resetPlant = () => {
        setPlantProgress(0);
        setConfig({ ...config, feedback: stateUtils.feedback.start, plantSize: 0 });
    };

    const handlePlantGrowth = () => {
        if (waterCoins > 0) {
            const newProgress = plantProgress + 1;
            if (newProgress >= POINTS.FULL_GROWN) {
                setConfig({
                    ...config,
                    feedback: stateUtils.feedback.win,
                }); //Change to the big trees pictures
            }
            onSpendPoint();
            setPlantProgress(newProgress);
            setConfig({ ...config, plantSize: getPlantSize() });
        } else {
            setConfig({ ...config, buttonState: false, feedback: stateUtils.feedback.noCoin });
        }
    };
    //show the progressbar
    const getProgress = () => {
        const progressPercentage = Math.round((plantProgress / POINTS.FULL_GROWN) * 100);
        return progressPercentage < 100 ? progressPercentage : 100;
    };

    return (
        <div className={styles.app}>
            <div className={styles.appHeader}>Grow your plant</div>

            <div>
                {`Tasks Completed: ${waterCoins}`}
                {/* plant display */}
                <Plant plantImage={stateUtils.plantImageUrls[getPlantSize()]}></Plant>
                {/* control buttons */}
                <ControlButton
                    control={stateUtils.controls.water}
                    points={1}
                    handlePlantGrowth={handlePlantGrowth}
                    image={stateUtils.buttonImageUrls.water}
                />
                <UserFeedback feedbackText={config.feedback} progress={getProgress()} />
                <ResetButton
                    control={stateUtils.controls.reset}
                    resetPlant={resetPlant}
                    image={stateUtils.buttonImageUrls.reset}
                />
                <div></div>
            </div>
        </div>
    );
}

export default PlantTreesModal;
