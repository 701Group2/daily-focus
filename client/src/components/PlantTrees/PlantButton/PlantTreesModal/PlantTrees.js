import React, { useState } from "react";
import Plant from "./PTMComponents/Plant";
import ControlButton from "./PTMComponents/ContolButton";
import UserFeedback from "./PTMComponents/UserFeedback";
import stateUtils from "../../stateUtils";
import "./PlantTrees.css";

export default function PlantTrees() {
    const [config, setConfig] = useState({
        plantSize: 0, //This var shows different trees pictures
        points: 0, //This var control the progressbar
        waterCoins: 30, // when the user complete one task they get some waterCoins, which can be used to water trees
        feedback: stateUtils.feedback.start,
    });
    const POINTS = {
        SEEDLING: 10, //small trees
        SMALL: 20, //medium trees
        FULL_GROWN: 51, // big trees
    };

    const handlePlantGrowth = () => {
        var coins = config.waterCoins;
        if (coins > 0) {
            const newPoints = config.points++;
            coins = coins - 1;
            setConfig({ ...config, points: newPoints, waterCoins: coins });
            if (newPoints >= POINTS.FULL_GROWN) {
                setConfig({ ...config, plantSize: 3, feedback: stateUtils.feedback.win }); //Change to the big trees pictures
            } else if (newPoints >= POINTS.SMALL) {
                setConfig({ ...config, plantSize: 2 }); //Change to the medium trees pictures
            } else if (newPoints >= POINTS.SEEDLING) {
                setConfig({ ...config, plantSize: 1 }); //Change to the small trees pictures
            }
        } else {
            coins = 0;
            setConfig({ ...config, feedback: stateUtils.feedback.noCoin });
        }
    };
    //show the progressbar
    const getProgress = () => {
        const progress = Math.round((config.points / POINTS.FULL_GROWN) * 100);
        return progress < 100 ? progress : 100;
    };

    return (
        <div className="App">
            <div className="App-header">Grow your plant</div>

            <div className="App-body">
                {/* plant display */}
                <Plant plantImage={stateUtils.plantImageUrls[config.plantSize]}></Plant>
                {/* control buttons */}

                <ControlButton
                    control={stateUtils.controls.water}
                    points={1}
                    handlePlantGrowth={handlePlantGrowth}
                    image={stateUtils.buttonImageUrls.water}
                />

                <UserFeedback feedbackText={config.feedback} progress={getProgress()} />

                <div></div>
            </div>
        </div>
    );
}
