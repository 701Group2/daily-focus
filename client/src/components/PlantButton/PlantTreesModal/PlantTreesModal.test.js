import React from "react";
import ShallowRenderer from "react-test-renderer/shallow";

import PlantTreesModal from ".";

test("it renders correctly", () => {
    const shallowRenderer = new ShallowRenderer();
    const snapshotComponent = shallowRenderer.render(<PlantTreesModal />);
    expect(snapshotComponent).toMatchSnapshot();
});
