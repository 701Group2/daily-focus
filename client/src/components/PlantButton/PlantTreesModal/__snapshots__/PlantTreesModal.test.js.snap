// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders correctly 1`] = `
<div
  className="app"
>
  <div
    className="appHeader"
  >
    Grow your plant
  </div>
  <div>
    Tasks Completed: undefined
    <Plant
      plantImage="https://user-images.githubusercontent.com/73207848/111860287-74fbbd00-89ab-11eb-808b-11a7216f1bb5.png"
    />
    <ControlButton
      control="water"
      handlePlantGrowth={[Function]}
      image="https://user-images.githubusercontent.com/73207848/111860325-a7a5b580-89ab-11eb-936d-bbc974cf89cf.png"
      points={1}
    />
    <UserFeedback
      feedbackText="When you finish a task, you will get water so you can water your tree! Try to click the watering can."
      progress={0}
    />
    <div />
  </div>
</div>
`;
